### ISSUE TITLE
`require-explicit-slots` does not recognize slot names enclosed in quotes

### ISSUE THREAD

**Issue by andreww2012** [vuejs/eslint-plugin-vue#2391]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.56.0
- **eslint-plugin-vue version:** 9.21.1
- **Vue version:** 3.4.15
- **Node version:** 18.18.0
- **Operating System:** Not relevant

**Please show your full configuration:**

Not relevant

**What did you do?**	
**What did you expect to happen?**
**What actually happened?**

`require-explicit-slots` does not recognize slot names enclosed in quotes. Sometimes the quotes are necessary to have a slot name with a hyphen. It falsely reports that the slot was not explicitly defined using `defineSlots`.

**Repository to reproduce this issue**

The simplest reproduction is just going to the [rule page on the eslint-plugin-vue site](https://eslint.vuejs.org/rules/require-explicit-slots.html#vue-require-explicit-slots) and enclose `foo` in any quotes like that:

```ts
defineSlots<{
  default(props: { msg: string }): any
  'foo'(props: { msg: string }): any
}>()
```

**Comment by so1ve** [comment]
Should close this right? :)

**Comment by FloEdelmann** [comment]
Yes, thanks for letting me know :slightly_smiling_face:
